subsys/bluetooth/host/gatt.c

```diff
@@ -638,7 +704,13 @@ static ssize_t cf_write(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 
 	bt_addr_le_copy(&cfg->peer, &conn->le.dst);
 	cfg->id = conn->id;
-	atomic_set_bit(cfg->flags, CF_CHANGE_AWARE);
+	set_change_aware(cfg, true);
```

Comment:
If a connected peer writes CF value before bonded and then reboots without disconnection, the written CF value is lost. On the other hand if the same peer disconnects before the reboot, the CF value is written to settings. Seems like bug.
Replicated this one with a Windows 10 used as Bluetooth Central.

Reply:
btw. Similar issue could affect CCC values if they are written before bonded.
Maybe we should store CCC and CF also on `pairing_complete` from `bt_conn_auth_info_cb `  to prevent the issue (I think that could be done in a separate PR)?

Reply:
Yeah I think we should only ever store to NVS when we are sure that we are bonded. I will double-check the spec to be sure.

Reply:
Discussed offline. We do not store to NVS when we are not bonded (we already have proper checks in code to ensure this).

The main problem is that we may not store the CCC and CF values if peer writes it and then bonds. We need to trigger write when bonding is complete (the `identity_resolved` callback would not be called if the connected peer does not use privacy).

Reply:
can you please open another bug and assign me?